operator/internal/deploy/chart_manager.go

```diff
@@ -0,0 +1,222 @@
+package deploy
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/kyma-project/lifecycle-manager/operator/api/v1alpha1"
+	"github.com/kyma-project/lifecycle-manager/operator/pkg/remote"
+	modulelib "github.com/kyma-project/module-manager/operator/pkg/manifest"
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	k8syaml "sigs.k8s.io/yaml"
+)
+
+var ErrExpectedExactlyOneSKRConfig = errors.New("expected exactly one SKR config")
+
+type SKRWebhookChartManager interface {
+	InstallWebhookChart(ctx context.Context, kyma *v1alpha1.Kyma,
```

Comment:
instead of passing the Sync Context, you could pass it into the struct during creation and not have to pass it as function arg in the interface.
Same goes for the cache

Reply:
I tried to pass it on creation and realised that it needs further refactoring.
Let's do it in [this](https://github.com/kyma-project/lifecycle-manager/issues/296) backlog item. 👍